File: codegolf/99bottlesofbeer.py
s=["%d bottle%s of beer","on the wall","Take one down and pass it around","Go to the store and buy some more"];i=99;t=['s']*101;t[1]='';b='.\n'
while i:a=s[0]%(i,t[i]);print a,s[1]+',',a+b+(s[3],s[2])[i>1]+",",s[0]%((i-1,99)[i<2],t[i-1]),s[1]+b;i-=1

File: codegolf/99bottlesofbeer.php
<?$i=99;$k='of beer';$b="bottles ".$k;$w="on the wall";$n="\n";
while($i-1){$z="$i $b $w, $i $b.{$n}Take one down and pass it around, ";
if($i==2){$p=$b;$b='bottle '.$k;}echo$z.--$i." $b $w.$n$n";}
$f="Go to the store and buy some more, 99 $p $w.";echo"1 $b $w, 1 $b.$n$f";